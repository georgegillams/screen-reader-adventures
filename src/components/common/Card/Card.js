import React from 'react';
import PropTypes from 'prop-types';
import { Card as GGCard } from 'gg-components/Card';
import HelperFunctions from 'helpers/common/HelperFunctions';
import nextifyHref from 'client-utils/nextifyHref';
import Link from 'next/link';

const Card = props => {
  const { href, onClick, ...rest } = props;

  const hrefExternal = href && HelperFunctions.includes(href, 'http');

  const renderNormalCard = !href || hrefExternal;

  const destination = nextifyHref(href);

  if (renderNormalCard) {
    return <GGCard href={href} onClick={onClick} {...rest} />;
  }

  return (
    <Link passHref href={destination.url} as={destination.as} {...destination.options}>
      <GGCard {...rest} />
    </Link>
  );
};

Card.propTypes = {
  href: PropTypes.string.isRequired,
  scroll: PropTypes.bool,
  onClick: PropTypes.func,
};

Card.defaultProps = {
  scroll: true,
  onClick: null,
};

export default Card;
