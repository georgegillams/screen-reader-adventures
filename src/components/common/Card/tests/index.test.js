import React from 'react';

import { render } from '@testing-library/react';

import Card from '..';

describe('<Card />', () => {
  it('Should render correctly', () => {
    const { container } = render(<Card href="/test" />);

    expect(container).toMatchSnapshot();
  });
  it('Should render with nextified href', () => {
    const { container } = render(<Card href="/blog/test" />);

    expect(container).toMatchSnapshot();
  });
  it('Should render correctly with custom className', () => {
    const { container } = render(<Card href="/test" className={'test-custom-classname'} />);

    expect(container).toMatchSnapshot();
  });
  it('Should render correctly hrefExternal', () => {
    const { container } = render(<Card href="https://duckduckgo.com/" />);

    expect(container).toMatchSnapshot();
  });
});
