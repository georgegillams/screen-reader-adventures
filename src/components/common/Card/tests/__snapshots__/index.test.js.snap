// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> Should render correctly 1`] = `
<div>
  <a
    class="card card--atomic card--not-disabled"
    href="/test"
  >
    <div
      class="card__outer-container"
    >
      <div
        class="card__background"
      />
      <div
        class="card__content-container card__content-container--padded"
      />
    </div>
  </a>
</div>
`;

exports[`<Card /> Should render correctly hrefExternal 1`] = `
<div>
  <a
    class="card card--atomic card--not-disabled"
    href="https://duckduckgo.com/"
  >
    <div
      class="card__outer-container"
    >
      <div
        class="card__background"
      />
      <div
        class="card__content-container card__content-container--padded"
      />
    </div>
  </a>
</div>
`;

exports[`<Card /> Should render correctly with custom className 1`] = `
<div>
  <a
    class="card test-custom-classname card--atomic card--not-disabled"
    href="/test"
  >
    <div
      class="card__outer-container"
    >
      <div
        class="card__background"
      />
      <div
        class="card__content-container card__content-container--padded"
      />
    </div>
  </a>
</div>
`;

exports[`<Card /> Should render with nextified href 1`] = `
<div>
  <a
    class="card card--atomic card--not-disabled"
    href="/blog/test"
  >
    <div
      class="card__outer-container"
    >
      <div
        class="card__background"
      />
      <div
        class="card__content-container card__content-container--padded"
      />
    </div>
  </a>
</div>
`;
